ansible/playbooks/roles/redpanda_broker/tasks/start-redpanda.yml

```diff
@@ -206,7 +206,7 @@
 #
 - name: Establish whether restart required
   set_fact:
-    restart_required: '("true" in restart_required_rc.stdout or (is_initialized and (nodeconfig_result.changed or "Removed" in package_result.results or "1 upgraded" in package_result.results))) and (restart_node | default("true") | bool)'
+    restart_required: '{{ ("true" in restart_required_rc.stdout or (is_initialized and (nodeconfig_result.changed or "Removed" in package_result.results or "1 upgraded" in package_result.results))) and (restart_node | default("true") | bool) }}'
```

Comment:
Somehow I'm managing to break the ansible runs with a debug statement that references a variable that (sometimes?) may not contain anything, and once it gets that way, future ansible runs break, even if the debug task is removed. I've tried running with `--flush-cache` on the ansible run and that's not fixing it.  What's also concerning is that once it breaks, even with removing the debug, it impacts the `Establish whether restart required` task.

```
TASK [redpanda_broker : Establish whether restart required] *********************************************************************************************************************************************************
task path: /Users/tcampbell/Documents/github/prs/pr-119-redpanda-data-deployment-automation/ansible/playbooks/roles/redpanda_broker/tasks/start-redpanda.yml:223
fatal: [XX.XX.XX.XX]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'results'. 'dict object' has no attribute 'results'\n\nThe error appears to be in '/Users/tcampbell/Documents/github/prs/pr-119-redpanda-data-deployment-automation/ansible/playbooks/roles/redpanda_broker/tasks/start-redpanda.yml': line 223, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n#\n- name: Establish whether restart required\n  ^ here\n"}
```

it looks like it might be something within the `package_result` variable.

Here's the example debug I was working with

```
- name: debug
  ansible.builtin.debug:
    msg:
    - "is_initialized is {{ is_initialized }}"
    - "restart_required_rc is {{ restart_required_rc.stdout }}"
    - "package_result is {{ package_result }}"
    - "nodeconfig_result is {{ nodeconfig_result }}"
    - "restart_node is {{ restart_node | default('true') | bool }}"
    - "is restart_required_rc.stdout true: {{ restart_required_rc.stdout }}"
    - "is nodeconfig_result changed {{ nodeconfig_result.changed }}"
    - "is package_result.results removed or ugprade {{ package_result.results }}"
#    - "what is_initialized and result of (nodeconfig_result.changed or package_result) {{ is_initialized and (nodeconfig_result.changed or 'Removed' in package_result.results or '1 upgraded' in package_result.results) }}"
#    - "restart_required: {{ ('true' in restart_required_rc.stdout or (is_initialized and (nodeconfig_result.changed or 'Removed' in package_result.results or '1 upgraded' in package_result.results))) and (restart_node | default('true') | bool) }}"
```

Reply:
Ok, per discussion earlier, we're going to ignore this issue for now since the underlying fix does make this resolve to true or false now.